src/emulator/apphosting/config.ts

```diff
@@ -1,75 +1,31 @@
 import { join } from "path";
 import { pathExists } from "fs-extra";
-import { readFileFromDirectory, wrappedSafeLoad } from "../../utils";
 import { logger } from "./utils";
 import { Emulators } from "../types";
+import { APPHOSTING_BASE_YAML_FILE } from "../../apphosting/config";
+import { AppHostingYamlConfig, loadAppHostingYaml } from "../../apphosting/yaml";
 
-type EnvironmentAvailability = "BUILD" | "RUNTIME";
-
-const APPHOSTING_YAML = "apphosting.yaml";
 const APPHOSTING_LOCAL_YAML = "apphosting.local.yaml";
 
-// Schema of apphosting.*.yaml files
-interface AppHostingYaml {
-  env?: {
-    variable: string;
-    secret?: string;
-    value?: string;
-    availability?: EnvironmentAvailability[];
-  }[];
-}
-
-interface AppHostingConfiguration {
-  environmentVariables?: Record<string, string>;
-  secrets?: Record<string, string>;
-}
-
-/**
- * Reads an apphosting.*.yaml file, parses, and returns in an easy to use
- * format.
- */
-export async function loadAppHostingYaml(
-  sourceDirectory: string,
-  fileName: string,
-): Promise<AppHostingConfiguration> {
-  const file = await readFileFromDirectory(sourceDirectory, fileName);
-  const apphostingYaml: AppHostingYaml = await wrappedSafeLoad(file.source);
-
-  const environmentVariables: Record<string, string> = {};
-  const secrets: Record<string, string> = {};
-
-  if (apphostingYaml.env) {
-    for (const env of apphostingYaml.env) {
-      if (env.value) {
-        environmentVariables[env.variable] = env.value;
-      }
-
-      if (env.secret) {
-        secrets[env.variable] = env.secret;
-      }
-    }
-  }
-
-  return { environmentVariables, secrets };
-}
-
 /**
  * Loads in apphosting.yaml & apphosting.local.yaml, giving
  * apphosting.local.yaml precedence if present.
  */
 export async function getLocalAppHostingConfiguration(
```

Comment:
small suggestion: if we make this environment-agnostic, this will help us provide apphosting linters for environments much more easily in the future, eg

getAppHostingConfiguration(environment)  // you can pass in 'staging' or 'local' for example